oap-server/server-storage-plugin/storage-iotdb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/iotdb/query/IoTDBEventQueryDAO.java

```diff
@@ -0,0 +1,184 @@
+                events.getEvents().add(parseEvent(event));
+            }
+        }
+        events.setTotal(storageDataList.size());
+        // resort by self, because of the select query result order by time.
+        final Order order = Objects.isNull(condition.getOrder()) ? Order.DES : condition.getOrder();
+        if (Order.DES.equals(order)) {
+            events.getEvents().sort(
+                    (org.apache.skywalking.oap.server.core.query.type.event.Event e1,
+                     org.apache.skywalking.oap.server.core.query.type.event.Event e2)
+                            -> Long.compare(e2.getStartTime(), e1.getStartTime()));
+        } else {
+            events.getEvents().sort(
+                    Comparator.comparingLong(org.apache.skywalking.oap.server.core.query.type.event.Event::getStartTime));
+        }
+        return events;
+    }
+
+    private StringBuilder whereSQL(final EventQueryCondition condition, StringBuilder query) {
+        query.append(" where 1=1");
+        if (!Strings.isNullOrEmpty(condition.getUuid())) {
+            query.append(" and ").append(Event.UUID).append(" = \"").append(condition.getUuid()).append("\"");
+        }
+        final Source source = condition.getSource();
+        if (source != null) {
+            if (!Strings.isNullOrEmpty(source.getService())) {
+                query.append(" and ").append(Event.SERVICE).append(" = \"").append(source.getService()).append("\"");
+            }
+            if (!Strings.isNullOrEmpty(source.getServiceInstance())) {
+                query.append(" and ").append(Event.SERVICE_INSTANCE).append(" = \"").append(source.getServiceInstance()).append("\"");
+            }
+            if (!Strings.isNullOrEmpty(source.getEndpoint())) {
+                query.append(" and ").append(Event.ENDPOINT).append(" = \"").append(source.getEndpoint()).append("\"");
+            }
+        }
+        if (!Strings.isNullOrEmpty(condition.getName())) {
+            query.append(" and ").append(Event.NAME).append(" = \"").append(condition.getName()).append("\"");
+        }
+        if (condition.getType() != null) {
+            query.append(" and ").append(Event.TYPE).append(" = \"").append(condition.getType().name()).append("\"");
+        }
+        final Duration time = condition.getTime();
+        if (time != null) {
+            if (time.getStartTimestamp() > 0) {
+                query.append(" and ").append(Event.START_TIME).append(" > ").append(time.getStartTimestamp());
+            }
+            if (time.getEndTimestamp() > 0) {
+                query.append(" and ").append(Event.END_TIME).append(" < ").append(time.getEndTimestamp());
+            }
+        }
+        // IoTDB doesn't support the query contains "1=1" and "*" at the meantime.
+        String queryString = query.toString();
+        queryString = queryString.replace("1=1 and ", "");
+        return new StringBuilder(queryString);
+    }
```

Comment:
the same

Reply:
A suggestion,
we can be the where conditions in a new StringBuilder to avoid operating string frequently. like following:
```java
StringBuilder where = new StringBuilder();
if (!Strigs.isNullOrEmpty(condition.getUuid())) {
    where.append(Event.UUID).append(" = \"").append(condition.getUuid()).append("\"").append(" and ");
}

if (where.length() > 0) {
    int length = where.length();
    where.delete(length - 4, length);
    query.append(" where ").append(where);
}

return query;
```

Reply:
Thanks for your suggestion. I will deal with it.